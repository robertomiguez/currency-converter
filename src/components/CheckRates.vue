<template>
    <div>
        <div class="wrap">
            <Money class='amount' v-model="amount" v-bind="money" :change="updateTickers(amount)"></Money>
            <Select class="select" :options="currencies" @selected="setSelected"/>
        </div>
        <div class='hint' v-show='amount===0'>Enter an amount to check the rates.</div>
    </div>
</template>

<script>
import { mapActions, mapGetters } from 'vuex'
import { Money } from 'v-money'
import Select from '../components/Select'

export default {
  name: 'CheckRates',
  components: {
    Money,
    Select
  },
  data: () => ({
    amount: 0,
    currency: 'BTC',
    money: {
      decimal: '.',
      thousands: ',',
      prefix: '',
      suffix: '',
      precision: 5,
      masked: false
    }
  }),
  methods: {
    ...mapActions([
      'updateTickers',
      'loadTickers'
    ]),
    async setSelected (selected) {
      try {
        this.currency = selected
        await this.loadTickers(this.currency)
        await this.updateTickers(this.amount)
      } catch (error) {
        this.images = []
      }
    }
  },
  computed: mapGetters({
    currencies: 'getCurrenciesId'
  }),
  mounted () {
    this.loadTickers(this.currency)
  }
}
</script>

<style>
.wrap {
  border: 0px;
  width: 350px;
  margin: 8px 0;
  display: inline-block;
  background-color: #f4f8fb;
}
.amount {
  margin-bottom: 15px;
  width: 250px;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  font-size: 28px;
  background-color: #f4f8fb;
  border-width: 0;
  outline:none;
}
.select {
    width: 50px;
}
.hint {
  font-size: 0.75em;
  color: darkgray;
  padding-bottom: 5px;
}

@media only screen and (max-width: 600px) {
  .wrap {
    width: 300px;
  }
  .amount {
    width: 200px;
  }
}

</style>
