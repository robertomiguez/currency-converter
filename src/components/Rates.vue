<template>
    <div>
        <div class="flex-grid" v-for="ticker in tickersCalculated" :key="ticker.pair">
            <div class="amount">{{ ticker.ask }}</div>
            <div class="flag">
                <img :src="require(`../assets/${ticker.pair.slice(3,6)}.png`)">
            </div>
            <div class="currency">
                {{ticker.pair.slice(3,6)}}
            </div>
        </div>
    </div>
</template>

<script>
import { mapGetters } from 'vuex'

export default {
  name: 'Rates',
  computed: mapGetters({
    tickersCalculated: 'getTickersCalculated'
  })
}
</script>

<style>
.flex-grid {
  display: flex;
  margin: 20px 0px;
  justify-content: center;
  align-items: center;
  margin-right: 100px;
}
.flex-grid .amount {
  width: 180px;
  text-align: center;
  font-size: 16px;
  background-color: #fff;
}
.flex-grid .flag {
  width: 50px;
  text-align: center;
  font-size: 16px;
}
.flex-grid .currency {
  width: 10px;
  text-align: center;
  font-size: 16px;
}
</style>
