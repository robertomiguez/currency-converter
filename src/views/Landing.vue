<template>
    <div class="content">
        <div class='title'>Currency Converter</div>
        <div class='description'>{{description}}</div>
        <CheckRates/>
        <Rates/>
    </div>
</template>

<script>
import CheckRates from '../components/CheckRates'
import Rates from '../components/Rates'

export default {
  name: 'Landing',
  components: {
    CheckRates,
    Rates
  },
  data: () => ({
    description: 'Receive competitive and transparent pricing with no hidden spreads. See how we compare.'
  })
}
</script>

<style>

.content {
  background-color: #fff;
  padding-top: 100px;
  height: 200px;
}
.title {
  font-size: 1.875em;
  padding-bottom: 20px;
}
.description {
  color: darkgray;
  text-align: center;
  padding-bottom: 25px;
  margin: auto;
  width: 22em;
}
</style>
