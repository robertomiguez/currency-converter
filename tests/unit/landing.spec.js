import { shallowMount } from '@vue/test-utils'
import Landing from '@/views/Landing'
import CheckRates from '@/components/CheckRates'
import Rates from '@/components/Rates'

describe('Landing', () => {
  const build = () => {
    const wrapper = shallowMount(Landing)

    return {
      wrapper,
      checkRates: () => wrapper.findComponent(CheckRates),
      rates: () => wrapper.findComponent(Rates)
    }
  }

  it('renders the component', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main child components', () => {
    const { checkRates, rates } = build()
    expect(checkRates().exists()).toBe(true)
    expect(rates().exists()).toBe(true)
  })

  it('passes a binded user prop to user profile component', () => {
    // arrange
    const { wrapper, checkRates } = build()
    wrapper.setData({
      amount: 0
    })

    // assert
    expect(checkRates().vm.amount).toBe(wrapper.vm.amount)
  })
})
