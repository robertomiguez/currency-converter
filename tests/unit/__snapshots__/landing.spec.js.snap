// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders the component 1`] = `
<div class="content">
  <div class="title">Currency Converter</div>
  <div class="description">Receive competitive and transparent pricing with no hidden spreads. See how we compare.</div>
  <checkrates-stub></checkrates-stub>
  <rates-stub></rates-stub>
</div>
`;
